### ISSUE TITLE
makedns and /var/named permissions

### ISSUE THREAD

**Issue by ChristopherWalker** [xcat2/xcat-core#5992]
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/deployment_guide/s2-bind-zone says:

/var/named/ "The working directory for the named service. The nameserver is not allowed to write to this directory. "

However xCAT changes those permissions to make this directory writeable. 

Before xCAT runs, the permissions are:

   [root@mgt-test ~]# ls -l /var | grep named
   drwxr-x---.  5 root named 4096 Feb  6 12:07 named

After makedns has run, the permissions are:

   [root@mgt-test ~]# ls -l /var | grep named
   drwxrwxr-x.  5 root named 4096 Feb  6 12:24 named

I suspect therefore that xCAT should be putting things in /var/named/dynamic/ rather than /var/named. 

Here's a transcript of a VM demonstrating this:

[root@mgt-test ~]# . /etc/profile.d/xcat.sh 
[root@mgt-test ~]# ls -l /var | grep named
drwxr-x---.  5 root named 4096 Feb  6 12:07 named
[root@mgt-test ~]# chdef -t group -o sdv ip='|sdv(\d+).*$|(sprintf("10.0.2.%d",($1+100)))|' 
[root@mgt-test ~]# nodeadd sdv[1-10] groups=all,sdv,compute,ipmi
[root@mgt-test ~]# ls -l /var | grep named
drwxr-x---.  5 root named 4096 Feb  6 12:07 named
[root@mgt-test ~]# makedns sdv[1-10]
Warning: SELINUX is not disabled. The makedns command will not be able to generate a complete DNS setup. Disable SELINUX and run the command again.
Handling sdv8 in /etc/hosts.
Handling sdv5 in /etc/hosts.
Handling sdv3 in /etc/hosts.
Handling sdv1 in /etc/hosts.
Handling sdv9 in /etc/hosts.
Handling sdv4 in /etc/hosts.
Handling sdv2 in /etc/hosts.
Handling sdv10 in /etc/hosts.
Handling sdv7 in /etc/hosts.
Handling sdv6 in /etc/hosts.
Getting reverse zones, this may take several minutes for a large cluster.
Completed getting reverse zones.
Updating zones.
Completed updating zones.
Restarting named
Restarting named complete
Updating DNS records, this may take several minutes for a large cluster.
Completed updating DNS records.
DNS setup is completed
[root@mgt-test ~]# ls -l /var | grep named
drwxrwxr-x.  5 root named 4096 Feb  6 12:24 named
[root@mgt-test ~]#

**Comment by immarvin** [comment]
hi @bybai , would you please take a look at this? thx

**Comment by bybai** [comment]
Hi @ChristopherWalker, thanks your pointing out this. `makedns` generates new`/etc/named.conf`, generates zone files named `db.*` under `/var/named`, restarts named service. 
By default, zone files directory is `/var/named`, `makedns` uses this default directory for years, the directory is configured in `/etc/named.conf`, the related doc is https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/deployment_guide/s1-bind#s2-bind-namedconf. 
In the past, we did not consider use another directories other than this default one, we will look into this carefully later. 
Since `makedns` will overwrite `/etc/named.conf`, if you have customized values, you need to back up related files before executing `makedns`.  Here I want to confirm: do you hit problem or conflicts here?

**Comment by ChristopherWalker** [comment]
Thanks for looking into this. 

We ran into this because our /var/named had ended up owned by root for reasons lost in the mists of time (perhaps a restore that overwrote the ownership). In the process of looking at this, I read the page from the RHEL documentation you quote - and the next one - where it says /var/named shouldn't be writeable by named. I'm not aware of the reasoning behind this though. 

Note that in ddns.pm the line:
     chmod 0775, $ctx->{dbdir}; # assure dynamic dns can actually execute against the directory
Changes the permissions on /var/named from the default of 0750


To put it another way, once we've restored the ownership, then I think things will work, so it won't be an operational issue for us.

**Comment by bybai** [comment]
Hi @ChristopherWalker, this bug is planned to fix. thanks your pointing out this.

**Comment by bybai** [comment]
Hi @ChristopherWalker, this is fixed in PR #6252, you can try the latest code. Thanks.
